internal/api/handlers/operations_handler.go

```diff
@@ -147,6 +144,64 @@ func (h *OperationsHandler) GetOperation(ctx context.Context, request *api.GetOp
 	return &api.GetOperationResponse{Operation: convertedOp}, nil
 }
 
+func (h *OperationsHandler) parseListOperationsResponse(ctx context.Context, operationEntities []*operation.Operation) ([]*api.ListOperationItem, error) {
+	operationResponse, err := requestadapters.FromOperationsToListOperationsResponses(operationEntities)
+	if err != nil {
+		h.logger.Error("error parsing operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	return operationResponse, nil
+}
+
+func (h *OperationsHandler) queryPendingOperations(ctx context.Context, schedulerName, sortingOrder string) ([]*operation.Operation, error) {
+	pendingOperationEntities, err := h.operationManager.ListSchedulerPendingOperations(ctx, schedulerName)
+	if err != nil {
+		h.logger.Error("error listing pending operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(pendingOperationEntities, sortingOrder)
+
+	return pendingOperationEntities, nil
+}
+
+func (h *OperationsHandler) queryActiveOperations(ctx context.Context, schedulerName, sortingOrder string) ([]*operation.Operation, error) {
+	activeOperationEntities, err := h.operationManager.ListSchedulerActiveOperations(ctx, schedulerName)
+	if err != nil {
+		h.logger.Error("error listing active operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(activeOperationEntities, sortingOrder)
+
+	return activeOperationEntities, nil
+}
+
+func (h *OperationsHandler) queryFinishedOperations(ctx context.Context, schedulerName, sortingOrder string, page, pageSize int64) ([]*operation.Operation, uint32, error) {
+	finishedOperationEntities, total, err := h.operationManager.ListSchedulerFinishedOperations(ctx, schedulerName, page, pageSize)
+	if err != nil {
+		h.logger.Error("error listing finished operations", zap.Error(err))
+		return nil, uint32(0), status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(finishedOperationEntities, sortingOrder)
+
+	return finishedOperationEntities, uint32(total), nil
+}
+
+func extractPaginationParameters(request *api.ListOperationsRequest) (uint32, uint32, error) {
+	page := request.GetPage()
+	pageSize := request.GetPerPage()
+	operationStage := request.Stage
+
+	if pageSize == 0 {
+		pageSize = 15
+	}
+
+	if operationStage != "final" && (request.Page != nil || request.PerPage != nil) {
+		return 0, 0, status.Errorf(codes.InvalidArgument, "there is no pagination filter implemented in : %s stage", operationStage)
+	}
+
+	return page, pageSize, nil
```

Comment:
Hey @mflilian the page will be extracted in human readable form, but it will be used as `array index`. Are you converting this value?